test/integration/shoots/applications/metrics.go

```diff
@@ -0,0 +1,130 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+	Overview
+		- Tests that runtime metrics are available
+
+	AfterSuite
+		- Cleanup Workload in Shoot
+
+	Test: Create arbitrary deployment
+	Expected Output
+		- Metrics are available
+ **/
+
+package applications
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"text/template"
+	"time"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	"github.com/gardener/gardener/test/framework"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/test/framework/resources/templates"
+	"github.com/onsi/ginkgo"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metricsv1beta1 "k8s.io/metrics/pkg/apis/metrics/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	metricsTestTimeout = 10 * time.Minute
+)
+
+var _ = ginkgo.Describe("Shoot application metrics testing", func() {
+
+	f := framework.NewShootFramework(&framework.ShootConfig{
+		CreateTestNamespace: true,
+	})
+
+	var (
+		name = "metrics-test"
+	)
+
+	f.Beta().CIt("should read runtime metrics", func(ctx context.Context) {
+		templateFilepath := filepath.Join(f.TemplatesDir, templates.SimpleLoadDeploymentName)
+		testTmpl, err := template.ParseFiles(templateFilepath)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Deploy the metrics test deployment")
+		var writer bytes.Buffer
+		err = testTmpl.Execute(&writer, map[string]string{
+			"name":      name,
+			"namespace": f.Namespace,
+		})
+		framework.ExpectNoError(err)
+
+		manifestReader := kubernetes.NewManifestReader(writer.Bytes())
+		err = f.ShootClient.Applier().ApplyManifest(ctx, manifestReader, kubernetes.DefaultMergeFuncs)
+		framework.ExpectNoError(err)
+
+		err = f.WaitUntilDeploymentIsReady(ctx, name, f.Namespace, f.ShootClient)
+		framework.ExpectNoError(err)
+
+		pods := &corev1.PodList{}
+		err = f.ShootClient.Client().List(ctx, pods, client.InNamespace(f.Namespace), client.MatchingLabels{"app": "load"})
+		framework.ExpectNoError(err)
+
+		if len(pods.Items) == 0 {
+			ginkgo.Fail("at least one pod is needed")
+		}
+		podName := pods.Items[0].Name
+
+		ginkgo.By("Check runtime metrics")
+		framework.ExpectNoError(
+			retry.Until(ctx, 30*time.Second, func(ctx context.Context) (bool, error) {
```

Comment:
Hm, maybe we should have a higher timeout here?

Reply:
It's the retry time, not a timeout. Do you still want to increase it?

Reply:
Hm, I can't see it in the code. For how long is this retried in case it fails?

Reply:
The timeout is configured [here](https://github.com/gardener/gardener/blob/8539fa54c882921a9b7f62bcfb947dc9516001aa/test/integration/shoots/applications/metrics.go#L124) and is set to 10 minutes. The context handed over by the test framework is be closed then and `retry.Until` won't continue.